### PR TITLE
Issue #1077 Consistency configuration

### PR DESCRIPTION
**Pull request by ljacomet**

### CODE REVIEW

File: clustered/common/src/main/java/org/ehcache/clustered/common/Consistency.java [hunk 1]
```diff
@@ -16,11 +16,20 @@
 package org.ehcache.clustered.common;
 
 /**
+ *
+ * Enumeration of the different consistency levels supported in clustered caches.
+ *
  * @author Ludovic Orban
```
Comment (GaryWKeim) [inline clustered/common/src/main/java/org/ehcache/clustered/common/Consistency.java]: These @author tags are to be removed, yes?

File: docs/src/docs/asciidoc/user/clustered-cache.adoc [hunk 2]
```diff
@@ -79,4 +79,36 @@ include::{sourcedir}/clustered/client/src/test/java/org/ehcache/clustered/client
...
+
+=== Specifying consistency level
+
+Ehcache offers two level of consistency:
+
+Eventual::
+In _eventual_ consistency, the clustering solution will make sure that when a mutative operation returns, invalidation for the corresponding entries found on other clients has been initiated.
+This is the default level if not configured explicitly.
+Strong::
+In _strong_ consistency, the clustering solution will make sure that when a mutative operation returns, the corresponding entries found on other clients have been invalidated.
```
Comment (AbfrmBlr) [inline docs/src/docs/asciidoc/user/clustered-cache.adoc]: Invalidation of keys on clients seems to be an internal thing, I think we should document strong vs eventual by explaining the visibility of change available to other clients.

File: clustered/common/src/main/java/org/ehcache/clustered/common/Consistency.java (hunk 1 above)
Comment (ljacomet) [inline clustered/common/src/main/java/org/ehcache/clustered/common/Consistency.java]: Indeed - removing

File: docs/src/docs/asciidoc/user/clustered-cache.adoc (hunk 2 above)
Comment (ljacomet) [inline docs/src/docs/asciidoc/user/clustered-cache.adoc]: That's a good point - will rephrase these and javadoc.

**Comment by ljacomet** [comment]
Updated following comments and also completed the Javadoc work.
Did also move the XML parsing classes to internal package.

**Comment by ljacomet** [comment]
test this please

File: clustered/client/src/main/java/org/ehcache/clustered/client/config/builders/ClusteredStoreConfigurationBuilder.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+package org.ehcache.clustered.client.config.builders;
+
+import org.ehcache.clustered.client.config.ClusteredStoreConfiguration;
+import org.ehcache.clustered.common.Consistency;
+import org.ehcache.config.Builder;
+
+/**
+ * {@link Builder} for the {@link ClusteredStoreConfiguration}.
+ */
+public class ClusteredStoreConfigurationBuilder implements Builder<ClusteredStoreConfiguration> {
```
Comment (albinsuresh) [inline clustered/client/src/main/java/org/ehcache/clustered/client/config/builders/ClusteredStoreConfigurationBuilder.java]: Do we really need a builder for this configuration? As there is only one parameter for the consistency configuration.

File: clustered/client/src/main/resources/ehcache-clustered-ext.xsd [hunk 4]
```diff
@@ -165,4 +165,17 @@
       </xs:annotation>
     </xs:attribute>
   </xs:attributeGroup>
+
+  <xs:element name="clustered-store" type="tc:clustered-store-type" substitutionGroup="eh:service-configuration"/>
```
Comment (albinsuresh) [inline clustered/client/src/main/resources/ehcache-clustered-ext.xsd]: Do we want to call this element clustered-"store"? Store is more of an internal concept, right? How about having the config something like this:
`<consistency>STRONG</consistency>` and `<consistency>EVENTUAL</consistency>` ?

File: clustered/client/src/test/java/org/ehcache/clustered/client/config/xml/ClusteringServiceConfigurationParserTest.java [hunk 5]
```diff
@@ -16,6 +16,7 @@
 
 package org.ehcache.clustered.client.config.xml;
 
+import org.ehcache.clustered.client.internal.config.xml.ClusteringServiceConfigurationParser;
```
Comment (albinsuresh) [inline clustered/client/src/test/java/org/ehcache/clustered/client/config/xml/ClusteringServiceConfigurationParserTest.java]: The parser moved to the internal package but the test did not.

File: clustered/client/src/main/java/org/ehcache/clustered/client/config/builders/ClusteredStoreConfigurationBuilder.java (hunk 3 above)
Comment (ljacomet) [inline clustered/client/src/main/java/org/ehcache/clustered/client/config/builders/ClusteredStoreConfigurationBuilder.java]: We have builders for all internal things - why make an exception?

File: clustered/client/src/main/resources/ehcache-clustered-ext.xsd (hunk 4 above)
Comment (ljacomet) [inline clustered/client/src/main/resources/ehcache-clustered-ext.xsd]: This is to be consistent with other store level configurations - have a look at the default Ehcache XSD.

File: clustered/client/src/test/java/org/ehcache/clustered/client/config/xml/ClusteringServiceConfigurationParserTest.java (hunk 5 above)
Comment (ljacomet) [inline clustered/client/src/test/java/org/ehcache/clustered/client/config/xml/ClusteringServiceConfigurationParserTest.java]: Good catch - will fix

**Comment by lorban** [comment]
Needs rebasing, otherwise LGTM

**Comment by lorban** [comment]
You may want to look at `DefaultClusteringService.getServerStoreProxy()` as it assumes the consistency can be null and sets the default to `EVENTUAL`. I don't like duplicating defaults so if one can go, that'as all for the best.

File: clustered/client/src/main/java/org/ehcache/clustered/client/config/builders/ClusteredStoreConfigurationBuilder.java (hunk 3 above)
Comment (albinsuresh) [inline clustered/client/src/main/java/org/ehcache/clustered/client/config/builders/ClusteredStoreConfigurationBuilder.java]: Configurations for serializers and copiers are exposed directly, no? How is this different from those cache level configurations?

**Comment by ljacomet** [comment]
@lorban Indeed, had missed that bit - fixed, including specifying a value in all the tests

**Pull request merged**